tools/core-build-tasks/src/tasks/bundle.ts

```diff
@@ -17,20 +21,105 @@ export type BundleTaskParameters = {
+): { [key: string]: string } {
+    const generatedFiles: { [key: string]: string } = {};
+    for (const element of outputFiles) {
+        if (element.path.endsWith(MAP_EXTENSION)) {
+            const parsedPath = path.parse(element.path);
+            const sourceMapFilePath = path.join(sourceMapDirectory, parsedPath.base);
+            const sourceMapContent = JSON.parse(element.text);
+
+            // Add JS file location.
+            sourceMapContent.file = path
+                .relative(sourceMapDirectory, path.join(outputDirectory, parsedPath.name))
+                .replace(/\\/g, '/');
+            generatedFiles[sourceMapFilePath] = JSON.stringify(sourceMapContent);
+        } else {
+            // Link to the source map file.
+            const dir = path.parse(element.path).dir;
+            const targetSourceMap = path
+                .join(path.relative(dir, sourceMapDirectory), path.parse(element.path).base)
+                .replace(/\\/g, '/');
+            generatedFiles[element.path] = element.text + `\n//# sourceMappingURL=${targetSourceMap}${MAP_EXTENSION}\n`;
+        }
+    }
+
+    return generatedFiles;
+}
+
+function writeFiles(postProcessOutputFilesResult: PostProcessOutputFilesResult) {
+    fs.mkdirSync(postProcessOutputFilesResult.outputDirectory, { recursive: true });
+    if (postProcessOutputFilesResult.sourceMapDirectory !== postProcessOutputFilesResult.outputDirectory) {
+        fs.mkdirSync(postProcessOutputFilesResult.sourceMapDirectory, { recursive: true });
+    }
+
+    for (const path of Object.keys(postProcessOutputFilesResult.generatedFiles)) {
+        fs.writeFileSync(path, postProcessOutputFilesResult.generatedFiles[path]);
+    }
+}
+
+export function postProcessOutputFiles(
+    options: BundleTaskParameters,
+    buildResult: BuildResult
+): PostProcessOutputFilesResult | undefined {
+    if (!buildResult.outputFiles) {
+        return undefined;
+    }
+
+    const outputDirectory = path.parse(options.outfile).dir;
+    const sourceMapDirectory = path.resolve(options.outputSourcemapPath ?? outputDirectory);
+    const generatedFiles = linkSourceMaps(sourceMapDirectory, outputDirectory, buildResult.outputFiles);
+    return { sourceMapDirectory, outputDirectory, generatedFiles: generatedFiles };
+}
+
 export function bundleTask(options: BundleTaskParameters): ReturnType<typeof parallel> {
-    return async () => {
-        return esbuild.build({
+    return () => {
+        const buildResult: BuildResult = esbuild.buildSync({
             entryPoints: [options.entryPoint],
             bundle: true,
             format: 'esm',
             minifyWhitespace: options.minifyWhitespace,
             outfile: options.outfile,
-            sourcemap: options.sourcemap,
+            sourcemap: options.sourcemap ? 'external' : false,
             external: options.external,
+            write: !options.sourcemap,
         });
+
+        if (buildResult.errors.length == 0) {
```

Comment:
nit: I don't enforce it in this repo (maybe I should?) but use `===` instead of `==`

Reply:
Fixed.
And, I think we should enforce it, it is a core library 😄

Reply:
Feel free to add me a work item for improvements :)